lib/utils/helpers.ts

```diff
@@ -292,6 +304,21 @@ export function getManualSortCompareFunction<T, U = T>(
 	};
 }
 
+/**
+ * Decide whether the current shell (that executed the CLI process) is a Windows
+ * 'cmd.exe' shell, including PowerShell, by checking a few environment
+ * variables.
+ */
+export function isWindowsComExeShell() {
+	return (
+		// neither bash nor sh (e.g. not MSYS, MSYS2, Cygwin, WSL)
+		process.env.SHELL == null &&
+		// Windows cmd.exe or PowerShell
+		process.env.ComSpec != null &&
+		process.env.ComSpec.endsWith('cmd.exe')
+	);
+}
```

Comment:
This is a refactoring of the `shellEscape` function. It was done during debugging. It turned out not to be relevant for the bug fixes, but it makes the the `shellEscape` function a bit easier to read.

Reply:
Very descriptive comments!

Reply:
Too much? :-) Occasionally I write more lines of comment than lines of code. :-D

Reply:
Nono, not too much!  Great IMO.